config/passport.js

```diff
@@ -15,14 +16,31 @@ module.exports = function(passport, models) {
         done(null, user);
     });
 
+    // Pass Google authentication configuration details
     passport.use(new GoogleStrategy({
         clientID: configAuth.googleAuth.clientID,
         clientSecret: configAuth.googleAuth.clientSecret,
         callbackURL: configAuth.googleAuth.callbackURL,
         access_type: configAuth.googleAuth.access_type
     },
+    /**
+     * Handle Google login using passport package
+     * @param  {Object}   req          Request object
+     * @param  {String}   accessToken  Google access token
+     * @param  {String}   refreshToken Google refresh token
+     * @param  {Object}   profile      Google profile data
+     * @param  {Function} done         Callback function
+     */
     function(req, accessToken, refreshToken, profile, done) {
         process.nextTick(function() {
+
+            if(profile.displayName.split(' ').length >= 2){
+                fname = profile.displayName.split(' ')[0];
+                lname = profile.displayName.split(' ')[1];
```

Comment:
I am using passport library for node js which handles Google authentication and it only provides retrieval of displayName property. Since passport handles the entire authentication, making google api calls seem redundant.